torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -870,6 +906,10 @@ def _reduce_grad(state: _FSDPState, handle: FlatParamHandle):
     return grad_to_offload
 
 
+def _run_all_reduce_as_async(state: _FSDPState, handle: FlatParamHandle) -> bool:
+    return True
```

Comment:
can we gate if we wanna land it

Reply:
Do we have any reason why we want to keep the existing non-async version?